pkg/apis/upgrade/v1alpha1/upgradeconfig_types.go

```diff
@@ -0,0 +1,164 @@
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// UpgradeConfigSpec defines the desired state of UpgradeConfig and upgrade window and freeze window
+type UpgradeConfigSpec struct {
+	// Specify the desired release
+	Desired       Update        `json:"desired"`
+	UpgradeWindow UpgradeWindow `json:"upgradeWindow"`
+	FreezeWindow  FreezeWindow  `json:"freezeWindow"`
+}
+
+// UpgradeConfigStatus defines the observed state of UpgradeConfig
+type UpgradeConfigStatus struct {
+
+	// +kubebuilder:validation:Enum={"New","Pending","Upgrading","Upgraded", "Failed"}
+	// This describe the status of the upgrade process
+	Phase UpgradePhase `json:"phase"`
+
+	Conditions []UpgradeCondition `json:"conditions,omitempty"`
+	// +kubebuilder:validation:Optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	CompleteTime *metav1.Time `json:"completeTime,omitempty"`
+}
+type UpgradeConditionType string
+
+const (
+	UpgradeValidated UpgradeConditionType = "Validated"
+	UpgradeComplete  UpgradeConditionType = "Completed"
```

Comment:
UpgradeCompleted

Reply:
I prefer not have the "Upgrade" prefix, We know it's talking about upgrade even don't have it, right?